crates/sui-types/src/crypto.rs

```diff
@@ -352,7 +352,7 @@ pub struct ConciseAuthorityPublicKeyBytes<'a>(&'a AuthorityPublicKeyBytes);
 
 impl std::fmt::Debug for ConciseAuthorityPublicKeyBytes<'_> {
     fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
-        let s = hex::encode(&self.0 .0[0..4]);
+        let s = encode_bytes_hex(&self.0 .0[0..4]);
```

Comment:
Taking the opportunity here. Per our recent SEV re panicking when accessing short slices or vectors, I'd personally add an `assert` to check the payload has enough bytes (at least 4) before doing`self.0[0..4]`. 

In particular for public keys, we know that current implementation is probably safe, but imagine what if @jonas-lj one day builds a naive Sig scheme where its pubKey is super short ie just 2-3 bytes.

That said, I'm opening an issue to check all subslice subvectors for panicking in all Sui's repos, starting from Sui, fastcrypto and MystenInfra. Lmk who wants to work on that.

Reply:
good idea!